pkg/osquery/runtime/runner.go

```diff
@@ -205,6 +253,13 @@ func (r *Runner) Query(query string) ([]map[string]string, error) {
 }
 
 func (r *Runner) Interrupt(_ error) {
+	if r.interrupted.Load() {
```

Comment:
Do you think it'd be useful to also do `r.rerunRequired.Store(false)` here? I'm thinking about the case where `UpdateRegistrationIDs` is called and then `Interrupt` is called while the shutdown/restart is ongoing. Could maybe add a test case for this as well?

Reply:
ohh yeah that seems wise. will do both!